scipy/signal/morebsplines.py

```diff
@@ -0,0 +1,544 @@
+    a quadratic spline:
+
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.signal import qspline1d, qspline1d_eval
+    >>> rng = np.random.default_rng()
+    >>> sig = np.repeat([0., 1., 0.], 100)
+    >>> sig += rng.standard_normal(len(sig))*0.05  # add noise
+    >>> time = np.linspace(0, len(sig))
+    >>> filtered = qspline1d_eval(qspline1d(sig), time)
+    >>> plt.plot(sig, label="signal")
+    >>> plt.plot(time, filtered, label="filtered")
+    >>> plt.legend()
+    >>> plt.show()
+
+    """
+    if lamb != 0.0:
+        raise ValueError("Smoothing quadratic splines not supported yet.")
+    else:
+        return _quadratic_coeff(signal)
+
+
+#pythran export cspline1d_eval(complex128[], float[])
+#pythran export cspline1d_eval(int[], int list)
+#pythran export cspline1d_eval(int[], int list, float)
+#pythran export cspline1d_eval(int[], int list, float, int)
+#pythran export cspline1d_eval(int[], float list)
+#pythran export cspline1d_eval(int[], float list, float)
+#pythran export cspline1d_eval(int[], float list, float, int)
+#pythran export cspline1d_eval(float[], int list)
+#pythran export cspline1d_eval(float[], int list, float)
+#pythran export cspline1d_eval(float[], int list, float, int)
+#pythran export cspline1d_eval(float[], float list)
+#pythran export cspline1d_eval(float[], float list, float)
+#pythran export cspline1d_eval(float[], float list, float, int)
+#pythran export cspline1d_eval(int[], int[])
+#pythran export cspline1d_eval(int[], int[], float)
+#pythran export cspline1d_eval(int[], int[], float, int)
+#pythran export cspline1d_eval(int[], float[])
+#pythran export cspline1d_eval(int[], float[], float)
+#pythran export cspline1d_eval(int[], float[], float, int)
+#pythran export cspline1d_eval(float[], int[])
+#pythran export cspline1d_eval(float[], int[], float)
+#pythran export cspline1d_eval(float[], int[], float, int)
+#pythran export cspline1d_eval(float[], float[])
+#pythran export cspline1d_eval(float[], float[], float)
+#pythran export cspline1d_eval(float[], float[], float, int)
+#pythran export cspline1d_eval(int[], int list, int)
+#pythran export cspline1d_eval(int[], int list, int, int)
+#pythran export cspline1d_eval(int[], float list, int)
+#pythran export cspline1d_eval(int[], float list, int, int)
+#pythran export cspline1d_eval(float[], int list, int)
+#pythran export cspline1d_eval(float[], int list, int, int)
+#pythran export cspline1d_eval(float[], float list, int)
+#pythran export cspline1d_eval(float[], float list, int, int)
+#pythran export cspline1d_eval(int[], int[], int)
+#pythran export cspline1d_eval(int[], int[], int, int)
+#pythran export cspline1d_eval(int[], float[], int)
+#pythran export cspline1d_eval(int[], float[], int, int)
+#pythran export cspline1d_eval(float[], int[], int)
+#pythran export cspline1d_eval(float[], int[], int, int)
+#pythran export cspline1d_eval(float[], float[], int)
+#pythran export cspline1d_eval(float[], float[], int, int)
```

Comment:
Whoa, there are a lot of signatures here. I suspect there's a more concise way of writing this - and if there isn't, we should create one. The pattern of having multiple signatures to express that a keyword may be given yes/no seems less than great.

Reply:
Or maybe the correct pattern is to leave the public function definition and docstring in Python, and only call the private Pythranized version - that way it's easy to guarantee that all arguments are positional.

Reply:
Will be an interesting discussion for our meeting tomorrow.